Rework intro and change "DP policy" section into "DAP integration" (#11)
wangshan/draft-wang-ppm-differential-privacy: draft-wang-ppm-differential-privacy.md

parties. Depending on the mechanism: the Clients might add noise to their own
measurements; the Aggregators might add noise to their aggregate shares (the
values they produce for the Collector); and the Collector might add noise to
the aggregate result before publishing it.

> NOTE The Collector adding noise is mentioned here only as a strawman. We
> typically consider the Collector to be under control of the attacker, so we
> want to ensure the aggregate result is DP before the Collector computes it.

> For some DP mechanisms, noise is added only by the Clients or only by the

Reviewer: mechanisms -> policies? If we are talking about noise by multiple parties here.

Author: Good catch. I also added a definition of "DP policy" above.

> For some DP mechanisms, noise is be added only by the Clients or only by the

Reviewer 2: ```suggestion
For some DP mechanisms, noise is added only by the Clients or only by the
```

   apply to VDAFs in general and DAP/VDAF in particular. Many refinements of
   the basic DP notion are possible, some of which are not possible given the
   constraints imposed by DAP. This section enumerates these constraints and
   describes the baseline DP requirements for DAP.

> 1. {{mechansims}} specifies various mechanisms required for building DP

Reviewer: ```suggestion
1. {{mechanisms}} specifies various mechanisms required for building DP
```

   systems, including algorithms for sampling from discrete Laplace and
   Gaussian distributions.

> 1. {{policies}} defines an "DP policies", which are cmposed with VDAFs in order

Reviewer: ```suggestion
1. {{policies}} defines "DP policies", which are composed with VDAFs in order
```

1. {{mechanisms}} defines an interface for DP mechanisms that are compatible
   with VDAFs and, for all VDAFs specified in {{!VDAF}} [TODO: And other
   drafts, once they appear], specifies concrete mechanisms.

> 1. {{dp-in-dap}} specifies the integration of DP mechanisms from the previous

Reviewer 2: I think there is still a concept of policy missing, a policy is implemented by DpMechanism but it may have concerns that’s not covered by DpMechanism, for eg. A policy of DP amplification depends on batch size, if it’s larger than min batch size, no noise needs to be added. I think this decision is the role of a policy type, which should be _used_ by DAP aggregators

Author: This falls into the bucket of changes for DAP. I suggested to Junye that we avoid mentioning protocl changes for now and just carve out a section in the draft for them. I do plan to mention this optmization in the PPM meeting however.

Author: Also, as discussed internally, we're going to merge this then rename "primitive" to "mechanism" and "mechanism" to "policy". Please plan on doing a full pass of the draft when you wake up and either file a PR to fix it up or drop comments somewhere :)

Author: Done.

   reduce the veracity of the result. This document provides no guidance for
   this selection process.

1. This document describes a particular class of narrowly-scoped DP policies.
   Other, more sophisticated policies are possible. [TODO: Add citations.]

> 1. The mechanisms described in {{mechansims}} are intended for use beyond

Reviewer: ```suggestion
1. The mechanisms described in {{mechanisms}} are intended for use beyond
```

> TODO: Chris P: we will mention Prio3SumVec because that's what we use to describe aggregator DP with amplification

> # DP Mechanisms {#mechansims}

Reviewer: ```suggestion
# DP Mechanisms {#mechanisms}
```

> TODO: Concretely describe an Aggregator-DP mechanism for Prio3Histogram.

# Executing DP Policies in DAP {#dp-in-dap}

> > TODO: Specify integration of a `DpMechanism` into DAP.

Reviewer: `
```suggestion
> TODO: Specify integration of a `DpPolicy` into DAP.
```
